src/main/scala/akka/persistence/jdbc/journal/dao/JournalTables.scala

```diff
@@ -27,14 +27,32 @@ trait JournalTables {
   def journalTableCfg: JournalTableConfiguration
 
   class Journal(_tableTag: Tag) extends Table[JournalRow](_tableTag, _schemaName = journalTableCfg.schemaName, _tableName = journalTableCfg.tableName) {
-    def * = (ordering, deleted, persistenceId, sequenceNumber, message, tags) <> (JournalRow.tupled, JournalRow.unapply)
+    def * = if (journalTableCfg.hasMessageColumn) {
+      (ordering, deleted, persistenceId, sequenceNumber, message, tags, event, eventManifest, serId, serManifest, writerUuid) <> (JournalRow.tupled, JournalRow.unapply)
+    } else {
+      (ordering, deleted, persistenceId, sequenceNumber, tags, event, eventManifest, serId, serManifest, writerUuid) <> ({
+        case (ordering, deleted, persistenceId, sequenceNumber, tags, event, eventManifest, serId, serManifest, writerUuid) =>
+          JournalRow(ordering, deleted, persistenceId, sequenceNumber, None, tags, event, eventManifest, serId, serManifest, writerUuid)
+      },
+        (row: JournalRow) => {
+          Some((row.ordering, row.deleted, row.persistenceId, row.sequenceNumber, row.tags, row.event, row.eventManifest, row.serId, row.serManifest, row.writerUuid))
+        })
+    }
 
     val ordering: Rep[Long] = column[Long](journalTableCfg.columnNames.ordering, O.AutoInc)
     val persistenceId: Rep[String] = column[String](journalTableCfg.columnNames.persistenceId, O.Length(255, varying = true))
     val sequenceNumber: Rep[Long] = column[Long](journalTableCfg.columnNames.sequenceNumber)
     val deleted: Rep[Boolean] = column[Boolean](journalTableCfg.columnNames.deleted, O.Default(false))
     val tags: Rep[Option[String]] = column[Option[String]](journalTableCfg.columnNames.tags, O.Length(255, varying = true))
-    val message: Rep[Array[Byte]] = column[Array[Byte]](journalTableCfg.columnNames.message)
+    @deprecated("This was used to store the the PersistentRepr serialized as is, but no longer.", "4.0.0")
+    private val message: Rep[Option[Array[Byte]]] = column[Array[Byte]](journalTableCfg.columnNames.message)
+
+    val event: Rep[Option[Array[Byte]]] = column[Option[Array[Byte]]](journalTableCfg.columnNames.event)
```

Comment:
As I understand it, the message column isn't included, because it's the composition of `*` that is used to know the schema, and that is controlled by whether `hasMessageColumn` is true or not. But the other columns are optional.

Note that in the case of Oracle, most of them need to be optional, since Oracle does not distinguish between null and empty.

Reply:
As far as I know, some people use these table definitions to generate the ddl. (as opposed to the pre-defined ddl's that we use in the tests).

It is a fair point that oracle does not distinguish between null and empty, however to keep the behaviour the same for all database we could explicitly make the columns `eventManifest` and `serManifest` optional, and insert None/null in case the manifests are empty.

This allows the non-legacy journal (and the row case class) to omit the old `message` and the `event`, `serId` and `writerUuid` can be made mandatory.

It might require a bit of fiddling to get this right (since this gives us two table definitions), but I think it is worth it